File: Java/Medium/WrongAnswer/minimum-number-of-food-buckets-to-feed-the-hamsters_0.java
class Solution {
    public int minimumBuckets(String hamsters) {
        
        int n = hamsters.length();
        int h = 0;
        for (int i = 0; i < n; i++) {
            if (hamsters.charAt(i) == 'H') {
                h++;
            }
        }
        int res = 0;
        int left = 0;
        int right = 0;
        for (int i = 0; i < n; i++) {
            if (hamsters.charAt(i) == 'H') {
                if (left == 0 && right == 0) {
                    left = 1;
                    right = 1;
                } else if (left == 0) {
                    left = right;
                    right = 1;
                } else if (right == 0) {
                    right = left;
                    left = 1;
                } else {
                    left = 1;
                    right = 1;
                }
            } else {
                if (left == 0 && right == 0) {
                    left = 1;
                    right = 1;
                } else if (left == 0) {
                    left = right;
                    right = 0;
                } else if (right == 0) {
                    right = left;
                    left = 0;
                } else {
                    left = 0;
                    right = 0;
                }
            }
            res += left + right;
        }
        if (h == 0) {
            return 0;
        }
        if (res < h) {
            return -1;
        }
        return res;
    }
}

File: C++/Medium/WrongAnswer/minimum-number-of-food-buckets-to-feed-the-hamsters_0.cpp
class Solution {
public:
    int minimumBuckets(string hamsters) {
        
        int n = hamsters.size();
        int res = 0;
        for (int i = 0; i < n; i++) {
            if (hamsters[i] == 'H') {
                if (i > 0 && hamsters[i - 1] == '.') {
                    res++;
                    hamsters[i - 1] = 'B';
                } else if (i < n - 1 && hamsters[i + 1] == '.') {
                    res++;
                    hamsters[i + 1] = 'B';
                } else {
                    return -1;
                }
            }
        }
        return res;
    }
};